File: Data Structures & Algorithms/anagram-groups/submission-0.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        let mp = new Map();
        for(let i = 0; i<strs.length;i++)
        {
            let s = strs[i].split("").sort().join("");
            if(mp.has(s))
            {
                mp.get(s).push(strs[i]);
            }
            else mp.set(s,[strs[i]]);
        }
        let ans = [];
        for(let [key,i] of mp)
        {
            ans.push(i);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/daily-temperatures/submission-0.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& arr) {
         vector<int>ans;
         int n = arr.size();
        //  int ans=0;
         for(int i=0;i<n;i++)
         {
            int c=0;
            int j = i+1;
            while(j<n && arr[i]>=arr[j]) j++;
            if(j<n)
            ans.push_back(j-i);
            else ans.push_back(0);
            
         }
         return ans;
    } 
};
